java/server/src/org/openqa/grid/common/RegistrationRequest.java

```diff
@@ -607,6 +607,10 @@ public void validate() throws GridConfigurationException {
           + HUB_HOST + " X -" + HUB_PORT + " 5555. The specified config was -" + HUB_HOST + " "
           + hub + " -" + HUB_PORT + " " + port);
     }
+    
+    if(port==-1){
```

Comment:
I think if they *don't* specify a port, we should assume grid default (which is 4444)